src/pykeen/nn/modules.py

```diff
@@ -43,17 +41,29 @@
     'UnstructuredModelInteraction',
 ]
 
+logger = logging.getLogger(__name__)
+
 
-def _ensure_tuple(*x: Union[Representation, Sequence[Representation]]) -> Tuple[Sequence[Representation], ...]:
-    return tuple(xx if isinstance(xx, Sequence) else (xx,) for xx in x)
+def _upgrade_to_sequence(x: Union[FloatTensor, Sequence[FloatTensor]]) -> Sequence[FloatTensor]:
+    return x if isinstance(x, Sequence) else (x,)
+
+
+def _ensure_tuple(*x: Union[Representation, Sequence[Representation]]) -> Sequence[Sequence[Representation]]:
```

Comment:
@cthoyt I am not sure how, but maybe we can get rid of the packing/unpacking sometimes necessary.

Reply:
yeah that's what I'm hoping to move towards. You saw that I externalized the `_get_batches()` function, so this `_ensure_tuple()` function is only used once. We need unit testssss